src/nidcpower/custom_types/lcr_measurement.py

```diff
@@ -0,0 +1,305 @@
+                self.ac_voltage_real = data.ac_voltage.real
+                self.ac_voltage_imaginary = data.ac_voltage.imag
+                self.ac_current_real = data.ac_current.real
+                self.ac_current_imaginary = data.ac_current.imag
+                self.z_real = data.z.real
+                self.z_imaginary = data.z.imag
+                self.z_magnitude, self.z_phase = data.get_z_magnitude_and_phase()
+                y = data.get_y()
+                self.y_real = y.real
+                self.y_imaginary = y.imag
+                self.y_magnitude, self.y_phase = data.get_y_magnitude_and_phase()
+                self.ls, self.cs, self.rs = data.series_lcr
+                self.lp, self.cp, self.rp = data.parallel_lcr
+                self.d = data.d
+                self.q = data.get_q()
+                self.measurement_mode = enums.InstrumentMode(data.measurement_mode).value
+                self.dc_in_compliance, self.ac_in_compliance = data.in_compliances
+                self.unbalanced = data.unbalanced
+            else:
+                for field_name, _ in struct_NILCRMeasurement._fields_:
+                    setattr(self, field_name, getattr(data, field_name))
+        else:
+            # Assign default value for all fields
+            for field_name, _ in struct_NILCRMeasurement._fields_:
+                setattr(self, field_name, 0)
+
+
+class LCRMeasurement(object):
+    """Specifies a LCR measurement."""
+    LCR = namedtuple(typename="LCR", field_names=("inductance", "capacitance", "resistance"))
+    InCompliances = namedtuple(typename="InCompliances", field_names=("dc", "ac"))
+
+    _lcr_measurement_field_metadata = [
+        # field_name           label(s)
+        ("channel"           , "Channel"                               ),  # noqa: E202,E203
+        ("vdc"               , "V DC"                                  ),  # noqa: E202,E203
+        ("idc"               , "I DC"                                  ),  # noqa: E202,E203
+        ("stimulus_frequency", "Stimulus frequency"                    ),  # noqa: E202,E203
+        ("ac_voltage"        , "AC voltage"                            ),  # noqa: E202,E203
+        ("ac_current"        , "AC current"                            ),  # noqa: E202,E203
+        ("z"                 , "Z"                                     ),  # noqa: E202,E203
+        ("series_lcr"        , ("Ls", "Cs", "Rs")                      ),  # noqa: E202,E203
+        ("parallel_lcr"      , ("Lp", "Cp", "Rp")                      ),  # noqa: E202,E203
+        ("d"                 , "D"                                     ),  # noqa: E202,E203
+        ("measurement_mode"  , "Measurement mode"                      ),  # noqa: E202,E203
+        ("in_compliances"    , ("DC in compliance", "AC in compliance")),  # noqa: E202,E203
+        ("unbalanced"        , "Unbalanced"                            ),  # noqa: E202,E203
+    ]
+
+    def __init__(
```

Comment:
There are two ways of constructing an LCRMeasurement here.
This is an object returned to the user by the driver exclusively, right?
Why do we need more than one way to construct this?

Reply:
Yes, this is an object returned to the user by the driver exclusively. As mentioned in https://github.com/ni/nimi-python/pull/1702#discussion_r920879292, I have removed all the unnecessary code paths, and now `LCRMeasurement` can only be constructed from an instance of `struct_NILCRMeasurement` in my latest commit.